homeassistant/components/mqtt/switch.py

```diff
@@ -126,7 +128,12 @@ def _setup_from_config(self, config):
         state_off = config.get(CONF_STATE_OFF)
         self._state_off = state_off if state_off else config[CONF_PAYLOAD_OFF]
 
-        self._optimistic = config[CONF_OPTIMISTIC]
+        self._optimistic = (
+            config[CONF_OPTIMISTIC] or config.get(CONF_STATE_TOPIC) is None
+        )
```

Comment:
Is this a bug fix, not directly related to this PR?

Reply:
For switch in fact we do not need an optimistic config option. With other platforms the absense of a state_topic is used to determine optimistic behaviour. This check was missing for the switch platform (which has only one state topic).
So this is indeed a bug as the current code base does not reflect the behavior we see with other platforms.
I want to suggest to deprecate `CONF_OPTIMISTIC` for this platform as it is implicit optimistic when a state topic is missing.